File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09 - While Loop Exercise/03. Vacation .py
needed_money = float(input())
owned_money = float(input())

total_days = 0
days_spend = 0

while owned_money < needed_money and days_spend != 5:
    command = input()
    money = float(input())
    total_days += 1
    if command == "save":
        owned_money += money
        days_spend = 0
    elif command == "spend":
        owned_money -= money
        days_spend += 1
        if owned_money < 0:
            owned_money = 0

if days_spend == 5:
    print("You can't save the money.")
    print(total_days)
if owned_money >= needed_money:
    print(f"You saved the money for {total_days} days.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/14. УПРАЖНЕНИЕ ЦИКЛИ - ЧАСТ 2/03. Vacation.js
function vacantion(data) {
    let needed = Number(data.shift());
    let money = Number(data.shift());

    let days = 0;
    let spendDays = 0;

    while ('Jessy has enugh money or spends money for 5 days in a row.') {
        let spendOrSave = data.shift();
        let amount = Number(data.shift());

        if (spendOrSave === 'save') {
            money += amount;
            spendDays = 0;
            days += 1;
            if (money >= needed) {
                console.log(`You saved the money for ${days} days.`);
                break;
            }
        } else if (spendOrSave === 'spend') {
            money -= amount;
            if (money < 0) {money = 0;}
            spendDays += 1
            days += 1
            if (spendDays == 5) {
                console.log(`You can't save the money.`);
                console.log(`${days}`)
                break;
            }
        }
    }
}

vacantion(['2000', '1000', 'spend', '1200', 'save', '2000']);
// Expected Output:
// You saved the money for 2 days.

vacantion(['110', '60', 'spend', '10', 'spend', '10', 'spend', '10', 'spend', '10', 'spend', '10']);
// Expected Output:
// You can't save the money.
// 5
